src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java

```diff
@@ -168,18 +168,18 @@ public ElasticsearchPersistentProperty getScoreProperty() {
 	public void addPersistentProperty(ElasticsearchPersistentProperty property) {
 		super.addPersistentProperty(property);
 
-		Parent annotation = property.findAnnotation(Parent.class);
+		if (property.isParentProperty()) {
+			ElasticsearchPersistentProperty parentProperty = this.parentIdProperty;
 
-		if (annotation != null) {
-			Assert.isNull(this.parentIdProperty, "Only one field can hold a @Parent annotation");
-			Assert.isNull(this.parentType, "Only one field can hold a @Parent annotation");
-			Assert.isTrue(property.getType() == String.class, "Parent ID property should be String");
-			this.parentIdProperty = property;
-			this.parentType = annotation.type();
-		}
+			if (parentProperty != null) {
+				throw new MappingException(
+						String.format("Attempt to add parent property %s but already have property %s registered "
+								+ "as version. Check your mapping configuration!", property.getField(), parentProperty.getField()));
+			}
 
-		if (property.isVersionProperty()) {
-			Assert.isTrue(property.getType() == Long.class, "Version property must be of type Long!");
+			Parent parentAnnotation = property.getType().getAnnotation(Parent.class);
```

Comment:
I'm wondering if that is actually intended here. What the implementation lookups is the `@Parent` annotation here:

```java
@Parent
class SomeType {}

class SomeOtherType {
  SomeType property;
}
```

where previously you would've discovered this:

```
class SomeOtherType {
  @Parent Object property;
}
```

I.e., where previously a property annotation was found we now lookup an annotation on the type of the property. Is that intentional?

Reply:
You're right. I've fix the issue and force-pushed the branch.